extensions/2.0/Khronos/KHR_materials_iridescence/README.md

```diff
@@ -0,0 +1,148 @@
+# KHR\_materials\_iridescence
+
+## Khronos 3D Formats Working Group
+
+* Pascal Schoen, adidas  
+* Ben Houston, threekit  
+* Mathias Kanzler, UX3D
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)  
+
+Copyright (C) 2018-2021 The Khronos Group Inc. All Rights Reserved. glTF is a trademark of The Khronos Group Inc.
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Exclusions
+
+* This extension must not be used on a material that also uses `KHR_materials_pbrSpecularGlossiness`.
+* This extension must not be used on a material that also uses `KHR_materials_unlit`.
+
+## Overview
+
+Iridescence describes an effect where hue varies depending on the viewing angle and illumination angle: A thin-film of a semi-transparent layer results in inter-reflections and due to thin-film interference, certain wavelengths get absorbed or amplified. Iridescence can be seen on soap bubbles, oil films, or on the wings of many insects.
+With this extension, thickness and index of refraction (IOR) of the thin-film can be specified, enabling iridescent materials. 
+
+## Extending Materials
+
+The iridescence materials are defined by adding the `KHR_materials_iridescence` extension to any glTF material.
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_iridescence": {
+                    "iridescenceFactor": 1.0,
+                    "iridescenceIOR": 1.8,
+                    "iridescenceThicknessMaximum": 1200.0
+                }
+            }
+        }
+    ]
+}
+```
+
+### Iridescence
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [appendix](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+|                              | Type                                                                | Description                                                                                      | Required              |
+|------------------------------|---------------------------------------------------------------------|--------------------------------------------------------------------------------------------------|-----------------------|
+|**iridescenceFactor**         | `number`                                                            | The iridescence intensity.                                                                       | No, default: `0.0`    |
```

Comment:
We discussed whether or not we could remove this factor and instead use thickness to fade out the effect. Is that possible with this model?

Reply:
Specifically, what's the difference between "intensity" dropping to zero vs. "thickness" dropping to zero?  Is there a visible difference, making it worthwhile to use two channels instead of a single combined channel?  (It would imply minimum locked to zero, of course).

Reply:
I just noticed the conversation of this on the mailing list... apologies for the duplication!

Reply:
Fot the non-mailing list users: The intensity is just an artistic control, so thickness is sufficient.